src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js

```diff
@@ -742,6 +816,47 @@ define([
       return Object.values(this.getFormData().associations).reduce((typeCount, typeItem) => {
         return typeCount + Object.values(typeItem).reduce((count, item) => count + item.length, 0);
       }, 0);
+    },
+
+    getUpdatedAssociations: function(product, associationType, productAndProductModelIdentifiers) {
+      const previousQuantifiedLinkedProducts = undefined !== product.quantified_associations && undefined !== product.quantified_associations[associationType] ?
+        product.quantified_associations[associationType].products :
+        [];
+      const previousQuantifiedLinkedProductModels = undefined !== product.quantified_associations && undefined !== product.quantified_associations[associationType] ?
+        product.quantified_associations[associationType].product_models :
+        [];
+
+      const linkedProducts = productAndProductModelIdentifiers.reduce((linkedProducts, identifier) => {
+        const matchProduct = identifier.match(/^product_(.*)$/);
+        if (!matchProduct) {
+          return linkedProducts;
+        }
+
+        const productIdentifierToAdd = matchProduct[1];
+        linkedProducts.filter((linkedProduct) =>  linkedProduct.identifier !== productIdentifierToAdd);
```

Comment:
I think : 
```suggestion
         linkedProducts = linkedProducts.filter((linkedProduct) =>  linkedProduct.identifier !== productIdentifierToAdd);
```

Same for product model

Reply:
Yes, this logic is actually outdated, we are currently working on it in RAC-23 ;)